core/src/main/java/com/scalar/db/common/StateManagedTwoPhaseCommitTransactionManager.java

```diff
@@ -23,34 +23,33 @@
 import com.scalar.db.exception.transaction.ValidationException;
 import java.util.List;
 import java.util.Optional;
-import java.util.concurrent.CopyOnWriteArrayList;
 
-public abstract class TransactionDecorationTwoPhaseCommitTransactionManager
-    extends AbstractTwoPhaseCommitTransactionManager
-    implements TwoPhaseCommitTransactionDecoratorAddable {
+public class StateManagedTwoPhaseCommitTransactionManager
```

Comment:
Ditto. Refactored this from using inheritance to using composition (decorator pattern).